File: src/main/java/com/runhwani/runmate/dto/response/group/GroupResponse.java
// src/main/java/com/runhwani/runmate/dto/GroupResponse.java
package com.runhwani.runmate.dto.response.group;

import lombok.*;

import java.time.OffsetDateTime;
import java.util.UUID;

/**
 * 그룹 생성 응답용 DTO
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class GroupResponse {
    /** 생성된 그룹 ID */
    private UUID groupId;
    /** 발급된 초대 코드 */
    private String inviteCode;
    /** 그룹명 */
    private String groupName;
    /** 연관 코스 ID */
    private UUID courseId;
    /** 시작 일시 */
    private OffsetDateTime startTime;
    /** 시작 위치 */
    private String startLocation;
    /** 시작 지점 위도 */
    private Double latitude;
    /** 시작 지점 경도 */
    private Double longitude;
}

File: RunMate/data/src/main/java/com/D107/runmate/data/remote/response/group/GroupResponse.kt
import com.D107.runmate.data.mapper.DataMapper
import com.D107.runmate.data.remote.common.BaseResponse
import com.D107.runmate.data.remote.response.group.GroupMemberResponse
import com.D107.runmate.data.remote.response.group.GroupMemberResponse.Companion.toDomainModel
import com.D107.runmate.domain.model.group.GroupData
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

@JsonClass(generateAdapter = true)
data class GroupResponse(
    @Json(name = "groupId")
    val groupId: String,
    @Json(name = "groupName")
    val groupName: String,
    @Json(name = "leaderId")
    val leaderId: String,
    @Json(name = "courseName")
    val courseName: String?,
    @Json(name = "courseId")
    val courseId: String?,
    @Json(name = "startTime")
    val startTime: String,
    @Json(name = "startLocation")
    val startLocation: String,
    @Json(name = "latitude")
    val latitude: Double,
    @Json(name = "longitude")
    val longitude: Double,
    @Json(name = "inviteCode")
    val inviteCode: String,
    @Json(name = "status")
    val status: Int,
    @Json(name = "members")
    val members: List<GroupMemberResponse>?
):BaseResponse{
    companion object : DataMapper<GroupResponse, GroupData> {
            override fun GroupResponse.toDomainModel(): GroupData {
            return GroupData(
                groupId = groupId,
                groupName = groupName,
                leaderId = leaderId,
                courseId = courseId,
                courseName = courseName,
                startTime = startTime,
                startLocation = startLocation,
                latitude = latitude,
                longitude = longitude,
                inviteCode = inviteCode,
                status = status,
                members = members?.map { it.toDomainModel() }?: emptyList()
            )

        }

    }
}